File: ListList.Api/Contracts/Partition.cs
namespace ListList.Api.Contracts;

public class Partition
{
    public Guid Id { get; set; }
    public string? Token { get; set; }

    public int Order { get; set; }

    public bool Checklist { get; set; } = false;
    public bool Owned { get; set; } = false;
    public bool Readonly { get; set; } = true;

    public string Label { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;

    public List<Node> Nodes { get; set; } = [];
    public List<ShareLink> ShareLinks { get; set; } = [];
}

File: ListList.Web/src/models/Partition.ts
import { Node, ShareLink } from '.';
import { ApiPartition } from '../contracts';

export interface Partition extends Omit<
  ApiPartition,
  'token' | 'nodes' | 'shareLinks'
> {
  checklist: boolean;
  isNotOwned?: boolean;
  nodes: Node[];
  pending: boolean;
  readonly: boolean;
  shareLinks: ShareLink[];
  tokens: string[] | null;
}
